src/video/wayland/SDL_waylandwindow.c

```diff
@@ -284,69 +285,83 @@ decoration_frame_configure(struct libdecor_frame *frame,
                            struct libdecor_configuration *configuration,
                            void *user_data)
 {
-    SDL_WindowData *wind = user_data;
+    SDL_WindowData *wind = (SDL_WindowData *)user_data;
     SDL_Window *window = wind->sdlwindow;
-    int width, height;
-    enum libdecor_window_state window_state;
     struct libdecor_state *state;
 
-    /* window size */
-    if (!libdecor_configuration_get_content_size(configuration, frame, &width, &height)) {
-        width = window->w;
-        height = window->h;
-    }
+    enum libdecor_window_state window_state;
+    int width, height;
 
-    Wayland_HandleResize(window, width, height, wind->scale_factor);
-    wind->shell_surface.libdecor.initial_configure_seen = SDL_TRUE;
+    SDL_bool focused = SDL_FALSE;
+    SDL_bool fullscreen = SDL_FALSE;
+    SDL_bool maximized = SDL_FALSE;
+    SDL_bool floating = libdecor_frame_is_floating(frame);
```

Comment:
Mind the undocumented libdecor behaviour: You have to call `libdecor_frame_is_floating` after `libdecor_frame_commit` because only then will the provided `window_state` be applied to the frame. I probably never encountered this, or forgot this, since I was always calling `libdecor_frame_is_floating` after `libdecor_frame_commit`, but we definitely have to document this in the API.

Alternatively, you can check `window_state` manually for non-floating states.

Reply:
Force-pushed to check window_state directly instead!